### PR TITLE
Children type checking

### LINKED ISSUE #726: Change logic for checking children types
We are currently using the `displayName` property for checking if children are of an specific type, for handling context based logic. 

This approach has been know to be fragile as its dependant on a string, often written by hand. Also it feels weird 🤪

@nicholasdalhaug Showed us a good way to improve this logic in [Banner](https://github.com/equinor/design-system/blob/72ff0768d01fe5c6bf8daee676355e740b05b9ac/libraries/core-react/src/Banner/Banner.tsx#L36)

Identified components which needs this logic changed:
* AccordianHeader
* Popover
* Menulist

Bonus: 
This will free us from the need to set `displayName` and let it be generated by functions names instead, as mentioned by @vnys in #721

### PR DESCRIPTION
**Pull request by mimarz**
resolves #726

* Removed usages of checking using `displayName === "some-components-displayName`
* Simplified `Popover` children splitting from its anchor element

### CODE REVIEW

File: libraries/core-react/src/Popover/Popover.tsx [hunk 1]
```diff
@@ -52,30 +60,22 @@ export const Popover = forwardRef<HTMLDivElement, Props>(function Popover(
...
-      if (child.type && child.type.displayName === 'eds-popover-anchor') {
-        anchorElement = child
-      } else {
-        // Add the remaining children to a new array to display inside <PopoverItem/>
-        childArray.push(child)
+
+  const { anchorElement, childArray } = React.Children.toArray(children).reduce(
+    (acc: PopoverSplit, child): PopoverSplit => {
+      if (isValidElement(child) && child.type === PopoverAnchor) {
+        return {
```
Comment (mimarz) [inline libraries/core-react/src/Popover/Popover.tsx:67]: using spread here resulted in empty `childArray` so reassignment like this had to be done instead... 🤷‍♂️

Does not work
```jsx
       return {
          anchorElement: child,
         ...acc,
        }
```

Works 🤯
```jsx
       return {
          anchorElement: child,
          childArray: acc.childArray,
        }
```

File: libraries/core-react/src/Popover/Popover.tsx [hunk 2]
```diff
@@ -52,30 +60,22 @@ export const Popover = forwardRef<HTMLDivElement, Props>(function Popover(
...
+          ...acc,
+          anchorElement: child,
+        }
```
Comment (pomfrida) [inline libraries/core-react/src/Popover/Popover.tsx:70]: Satisfying ❤️

**Review by pomfrida — APPROVED** [review]
LGTM 👍

**Review by wenche — APPROVED** [review]
LGTM 👍

**Pull request merged**